azext_iot/iothub/_help.py

```diff
@@ -309,3 +309,50 @@ def load_iothub_help():
         type: command
         short-summary: Upload a local file as a device to a pre-configured blob storage container.
     """
+
+    helps["iot edge devices"] = """
+        type: group
+        short-summary: Commands to create and manage IoT Edge devices.
+    """
+
+    helps["iot edge devices create"] = """
+        type: command
+        short-summary: Create configuratations of edge devices in an IoT Hub.
+        long-summary: |
+          This operation accepts inline device arguments or an edge devices configuration file in YAML or JSON format.
+          Inline command args (like '--device-auth') will take precedence and override configuration file properties if they are provided.
+          A sample configuration file can be found here: https://github.com/Azure/azure-iot-cli-extension/tree/dev/docs/samples/sample_devices_config.yaml
+          Review examples and parameter descriptions for details on how to fully utilize this operation.
+
+        examples:
+        - name: Create a couple of edge devices using symmetric key auth (default)
+          text: |
+            az iot edge devices create -n {hub_name} --device id=device_1 --device id=device_2
+
+        - name: Create a flat list of edge devices using self-signed certificate authentication with various edge property configurations, using inline arguments.
+          text: |
+            az iot edge devices create -n {hub_name} --device-auth x509_thumbprint --default-edge-agent {default_edge_agent}
+            --device id=device_1 hostname={FQDN}
+            --device id=device_2 edge_agent={custom_agent}
+            --device id=parent hostname={FQDN} edge_agent={custom_agent} container_auth={path_or_json_string}
+
+        - name: Delete all existing devices on a hub and create new devices based on a configuration file (with progress bars and visualization).
```

Comment:
cant we just tell the user "run the delete command first" (this will require creating said delete command) rather than slapping this into create and making it have more switches/toggles?

Reply:
For the sake of this command, since it's being used to create specifically designated device scenarios, I would argue it's acceptable to have a `--clean` switch here. That's not to say we shouldn't also eventually have a `devices clean` command, but I don't think adding one extra switch to a pretty hefty "infra as config" type command complicates it too much